src/modules/user/services/user.esdt.service.ts

```diff
@@ -11,88 +10,65 @@ import { TokenService } from 'src/modules/tokens/services/token.service';
     ) {}
 
-    @GetOrSetCache({
-        baseKey: 'user',
-        remoteTtl: Constants.oneSecond() * 6,
-    })
-    private async uniquePairTokens(): Promise<string[]> {
-        return await this.getUniquePairTokensRaw();
-    }
-
-    private async getUniquePairTokensRaw(): Promise<string[]> {
-        const promises = [];
-        const [pairsAddresses, uniquePairTokens] = await Promise.all([
-            this.routerAbi.pairsAddress(),
-            this.tokenService.getUniqueTokenIDs(false),
-        ]);
-
-        for (const address of pairsAddresses) {
-            promises.push(this.pairAbi.lpTokenID(address));
-        }
-
-        const lpTokensIDs = await Promise.all(promises);
-        uniquePairTokens.push(...lpTokensIDs);
-
-        return uniquePairTokens;
-    }
-
     async getAllEsdtTokens(
         userAddress: string,
         pagination: PaginationArgs,
         inputTokens?: IEsdtToken[],
     ): Promise<UserToken[]> {
-        const [userTokens, uniquePairTokens] = await Promise.all([
-            inputTokens
-                ? Promise.resolve(inputTokens)
-                : this.apiService.getTokensForUser(
-                      userAddress,
-                      pagination.offset,
-                      pagination.limit,
-                  ),
-            this.uniquePairTokens(),
-        ]);
-
-        const userPairEsdtTokens = userTokens.filter((token) =>
-            uniquePairTokens.includes(token.identifier),
+        const [userTokens, uniquePairTokens, pairsAddresses] =
+            await Promise.all([
+                inputTokens
+                    ? Promise.resolve(inputTokens)
+                    : this.apiService.getTokensForUser(
+                          userAddress,
+                          pagination.offset,
+                          pagination.limit,
+                      ),
+                this.tokenService.getUniqueTokenIDs(false),
+                this.routerAbi.pairsAddress(),
+            ]);
+        const lpTokensIDs = await this.pairService.getAllLpTokensIds(
+            pairsAddresses,
         );
-        const promises = userPairEsdtTokens.map((token) => {
-            return this.getEsdtTokenDetails(new EsdtToken(token));
-        });
 
-        return await Promise.all(promises);
-    }
+        const userPairEsdtTokens = userTokens
+            .filter((token) => uniquePairTokens.includes(token.identifier))
+            .map((token) => new EsdtToken(token));
 
-    private async getEsdtTokenDetails(token: EsdtToken): Promise<UserToken> {
-        const pairAddress = await this.pairService.getPairAddressByLpTokenID(
-            token.identifier,
-        );
+        const userLpEsdtTokens = userTokens
+            .filter((token) => lpTokensIDs.includes(token.identifier))
+            .map((token) => new EsdtToken(token));
+
+        const orderedPairAddresses = [];
+        userLpEsdtTokens.forEach((token, index) => {
```

Comment:
this `forEach` is hard to understand why you need a list of ordered pair addresses

Reply:
The call `this.userEsdtCompute.allLpTokensUSD(..)` accepts an array of esdt tokens, and an array of pair addresses that need to match the LP tokens.